mmv1/products/vertexai/api.yaml

```diff
@@ -103,6 +106,220 @@ objects:
         input: true
         description: |
           Points to a YAML file stored on Google Cloud Storage describing additional information about the Dataset. The schema is defined as an OpenAPI 3.0.2 Schema Object. The schema files that can be used here are found in gs://google-cloud-aiplatform/schema/dataset/metadata/.
+
+# Vertex AI Endpoints
+  - !ruby/object:Api::Resource
+    name: Endpoint
+    base_url: projects/{{project}}/locations/{{location}}/endpoints
+    create_url: projects/{{project}}/locations/{{location}}/endpoints?endpointId={{name}}
+    self_link: 'projects/{{project}}/locations/{{location}}/endpoints/{{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/vertex-ai/docs'
+      api: 'https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations.endpoints'
+    async: !ruby/object:Api::OpAsync
+      actions:
+        - create
+        - delete
+      operation: !ruby/object:Api::OpAsync::Operation
+        path: 'name'
+        base_url: '{{op_id}}'
+        wait_ms: 1000
+      result: !ruby/object:Api::OpAsync::Result
+        path: 'response'
+        resource_inside_response: true
+      status: !ruby/object:Api::OpAsync::Status
+        path: 'done'
+        complete: True
+        allowed:
+          - True
+          - False
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error'
+        message: 'message'
+    description: "Models are deployed into it, and afterwards Endpoint is called to obtain predictions and explanations."
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: location
+        description: The location for the resource
+        url_param_only: true
+        required: true
+        input: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: name
+        description: The resource name of the Endpoint.
+        url_param_only: true
+        required: true
+        input: true
+      - !ruby/object:Api::Type::String
+        name: displayName
+        description: Required. The display name of the Endpoint. The name can be up to 128 characters long and can consist of any UTF-8 characters.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: description
+        description: The description of the Endpoint.
+      - !ruby/object:Api::Type::Array
+        name: deployedModels
+        description: Output only. The models deployed in this Endpoint. To add or remove DeployedModels use EndpointService.DeployModel and EndpointService.UndeployModel respectively.
```

Comment:
Can you update the description here? It references API endpoints that the user  probably won't directly interact with. Maybe link a guide to updating deployedModels?

Reply:
I couldn't find such a guide in the vertex docs, so I added a link to the cloud console instead. I will also add a short description of the model deployment resource here once it is implemented.

Reply:
Great!